File: editor/en/[621]Task Scheduler.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] arr = new int[26];

        for (char task : tasks) {
            arr[task - 'A']++;
        }

        int maxFreq = 0;

        for (int freq : arr) {
            maxFreq = Math.max(maxFreq, freq);
        }

        int maxCount = 0;

        for (int freq : arr) {
            if (freq == maxFreq) {
                maxCount++;
            }
        }

        int emptySlots = (maxFreq - 1) * (n - maxCount + 1);
        int availableTasks = tasks.length - maxFreq * maxCount;
        int idleSlots = Math.max(0, emptySlots - availableTasks);
        return tasks.length + idleSlots;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[621]Task Scheduler.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        arr = [0] * 26

        for task in tasks:
            arr[ord(task) - ord('A')] += 1;

        maxFreq = max(arr)
        maxCount = arr.count(maxFreq)

        emptySlots = (maxFreq - 1) * (n - maxCount + 1)
        availableTasks = len(tasks) - maxFreq * maxCount
        idleSlots = max(0, emptySlots - availableTasks)
        return len(tasks) + idleSlots

# leetcode submit region end(Prohibit modification and deletion)
